### DISCUSSION TITLE
Zod Naming

### DISCUSSION THREAD

**Discussion by s-cork** [anvilistas/anvil-labs#120] [General]
tagging @hugetim 

I've been playing around with python typing
Not something I've done much of
But I'm wondering if we should consider changing the names of things to match python's typing module

```python

z.record   -> z.mapping     # Python has types.Mapping: Mapping[str, int]
z.object   -> z.typeddict   # Python has types.TypedDict: TypeDict("Point", {"x": int, "y": int})
z.optional -> z.notrequired # Python has types.NotRequired type which can be applied to TypedDict types 
                            # TypeDict("Foo", {"x1": int, "x2": int, "x3": NotRequired[int]})
z.nullable -> z.optional    # types.Optional[str] means str | None
z.array    -> z.list


```

I guess that would mean we would end up with something like
```python
user_schema = z.typeddict(
    {
        "name": z.string().strip().min(3).max(50).transform(str.capitalize),
        "email": z.string().strip().email(),
        "age": z.number(invalid_type_error="Expected a number").min(18).max(100).notrequired(),
    }
)
```

I guess a motivation for this would be that the goal of typescript zod is to be like for like with typescript typing
maybe this implementation should be a like for like with python typing

But maybe I'm just overthinking this and actually it's better to stay closer to the original zod api

**Comment by hugetim** [comment]
I'm even less familiar, but that all makes sense to me. Following Python typing seems better, for sure. It seems ideal to make Anvil as pythonic as possible.

Maybe change the module name to pyzod to avoid confusing those familiar with the original zod?

**Reply by InjAnnuity1** [reply]
I second the pyzod moniker.  PYPI already has
https://pypi.org/project/zod/
but no pyzod.
